dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/boot/internal/KernelBootstrap.java

```diff
@@ -523,25 +528,58 @@ private static String constructJSONHeader(String consoleLogHeader, BootstrapConf
         String serverHostName = getServerHostName();
         String datetime = getDatetime();
         String sequenceNumber = getSequenceNumber();
-        //construct json header
-        StringBuilder sb = new StringBuilder();
-        sb.append("{\"type\":\"liberty_message\"");
-        sb.append(",\"host\":\"");
-        sb = jsonEscape(sb, serverHostName);
-        sb.append("\",\"ibm_userDir\":\"");
-        sb = jsonEscape(sb, wlpUserDir);
-        sb.append("\",\"ibm_serverName\":\"");
-        sb = jsonEscape(sb, serverName);
-        sb.append("\",\"message\":\"");
-        sb = jsonEscape(sb, consoleLogHeader);
-        sb.append("\",\"ibm_datetime\":\"");
-        sb = jsonEscape(sb, datetime);
-        sb.append("\",\"ibm_sequence\":\"");
-        sb = jsonEscape(sb, sequenceNumber);
-        sb.append("\"}");
-
-        return sb.toString();
 
+        //header field names and values
+        List<String> headerFieldNames = new ArrayList<>(Arrays.asList("type", "host", "ibm_userDir", "ibm_serverName", "message", "ibm_datetime", "ibm_sequence"));
+        final String[] headerFieldValues = { "liberty_message", serverHostName, wlpUserDir, serverName, consoleLogHeader, datetime, sequenceNumber };
+
+        final String OMIT_FIELDS_STRING = "@@@OMIT@@@";
+        Boolean omitJsonFields = Boolean.valueOf(System.getenv("WLP_LOGGING_OMIT_JSON_FIELD_MAPPINGS"));
+
+        //bootstrap fieldMappings should take precedence
+        String bsFieldMappings = bootProps.get("com.ibm.ws.logging.json.field.mappings");
+        String envFieldMappings = System.getenv("WLP_LOGGING_JSON_FIELD_MAPPINGS");
+        String fieldMappings = bsFieldMappings != null ? bsFieldMappings : envFieldMappings;
+
+        String[] keyValuePairs = fieldMappings.split(",");
+        for (String pair : keyValuePairs) {
+            pair = pair.trim();
+            if (pair.endsWith(":") && omitJsonFields) //omitJsonFields beta guard
+                pair = pair + OMIT_FIELDS_STRING;
+
+            String[] entry = pair.split(":");
+            entry[0] = entry[0].trim();
+
+            if (headerFieldNames.contains(entry[entry.length - 2])) {
+                if (entry.length == 2 && !pair.endsWith(":")) {
+                    entry[1] = entry[1].trim();
+                    headerFieldNames.set(headerFieldNames.indexOf(entry[0]), entry[1]);
```

Comment:
need to check if entry[0] is in headerFieldNames first.  if it isn't, and you call headerFieldNames.set(-1,...) you'll get an exception.